internal/config/config.go

```diff
@@ -226,7 +226,7 @@ type MTLS struct {
 // NamespaceSelector contains config options for filtering namespaces.
 type NamespaceSelector struct {
 	// MatchLabels is a list of labels to filter namespaces with.
-	MatchLabels map[string]string `mapstructure:"matchLabels"`
+	MatchLabels map[string]interface{} `mapstructure:"matchLabels"`
```

Comment:
Could we leverage from metav1:
```
// A label selector is a label query over a set of resources. The result of matchLabels and
// matchExpressions are ANDed. An empty label selector matches all objects. A null
// label selector matches no objects.
// +structType=atomic
type LabelSelector struct {
	// matchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels
	// map is equivalent to an element of matchExpressions, whose key field is "key", the
	// operator is "In", and the values array contains only "value". The requirements are ANDed.
	// +optional
	MatchLabels map[string]string `json:"matchLabels,omitempty" protobuf:"bytes,1,rep,name=matchLabels"`
	// matchExpressions is a list of label selector requirements. The requirements are ANDed.
	// +optional
	MatchExpressions []LabelSelectorRequirement `json:"matchExpressions,omitempty" protobuf:"bytes,2,rep,name=matchExpressions"`
}
```

also in the package labels there is something similar

Reply:
we use it for example [here](https://github.com/newrelic/newrelic-infra-operator/blob/a945b9a1b6d9f0f0f7065c97d5978dc0c4041acf/internal/mutator/pod/agent/injector.go#L86)

(I know that this PR as a different scope, I just noticed it now)